### ISSUE TITLE
Return data from model to controller

### ISSUE THREAD

**Issue by narangrishab** [balderdashy/sails#3333]
Hello,

Controller: User.getUserDetails(req,res,cb)

Model:

 getUserDetails: function(req, res, cb){

  ..
  ..

 cb(false, data);  //This gives error
  
}

error: 

{
error: "E_UNKNOWN",
status: 500,
summary: "Encountered an unexpected error",
raw: {
code: "ER_BAD_FIELD_ERROR",
errno: 1054,
sqlState: "42S22",
index: 0
}
}

On doing console.log() I get : ER_BAD_FIELD_ERROR: Unknown column 'NaN' in 'where clause'
even though there's no where condition

Even though all the fields exist properly I get this error.

cb(false); -> returns the same thing

cb(data) returns 
<img width="454" alt="screen shot 2015-10-27 at 2 57 41 pm" src="https://cloud.githubusercontent.com/assets/9140727/10754060/1a1800d6-7cbb-11e5-8944-28f3ba1e12f4.png">

**Comment by sailsbot** [comment]
Thanks for posting, @narangrishab.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
